.azure/pipelines/jobs/default-build.yml

```
@@ -144,7 +147,7 @@ jobs:
   - ${{ if eq(parameters.artifacts.publish, 'true') }}:
     - task: PublishBuildArtifacts@1
       displayName: Upload artifacts
-      condition: eq(variables['system.pullrequest.isfork'], false)
+      condition: or(eq(variables['system.pullrequest.isfork'], false), eq('${{ parameters.artifacts.force }}', 'true'))
```

```suggestion
      condition: or(eq(variables['system.pullrequest.isfork'], false), eq(parameters.artifacts.force, 'true'))
```
Can be simplified a little.

---

This doesn't work, I tried two times already. unknown `parameters` or something similar. I can point you to the specific builds if you want to see.

---

I'm surprised it doesn't work. Two lines up we have this: `- ${{ if eq(parameters.artifacts.publish, 'true') }}:`

---

https://dev.azure.com/dnceng/public/_build/results?buildId=76505

---

You're right. Got confused by the yml syntax again. This is one of the pitfalls of the double-pass evaluation. Contents inside `${{ }}` are evaluated by the template engine. `condition` are evaluated at "runtime" when the build executes.

The way you have it now is fine.